File: 14-TypeScript/Grey Multi-Tenant/packages/grey-julia/src/UserClient.jl
"""
User client for the Grey SDK.
"""
module UserClient

export get_user, get_current_user

using ..HttpClient
using ..GreyError
using ..Results

"""
    get_user(client::HttpClient.Client, user_id::String) -> Result

Get a user by ID.
"""
function get_user(client::HttpClient.Client, user_id::String)
    # Validate inputs
    if isempty(strip(user_id))
        return Results.err(GreyError.validation("User ID is required"))
    end
    
    HttpClient.get(client, "/users/$(user_id)")
end

"""
    get_current_user(client::HttpClient.Client) -> Result

Get the current authenticated user.
"""
function get_current_user(client::HttpClient.Client)
    HttpClient.get(client, "/users/me")
end

end # module

File: 14-TypeScript/Grey Multi-Tenant/packages/grey-matlab/+grey/+domain/UserClient.m
classdef UserClient < handle
    %USERCLIENT User client for Grey SDK
    %   Handles user-related operations.
    
    properties (SetAccess = private)
        httpClient  % grey.http.HttpClient
    end
    
    methods
        function obj = UserClient(httpClient)
            %USERCLIENT Construct user client
            %   client = UserClient(httpClient)
            
            obj.httpClient = httpClient;
        end
        
        function result = get(obj, userId)
            %GET Get a user by ID
            %   result = client.get(userId)
            
            % Validate inputs
            if isempty(strtrim(userId))
                result = grey.error.Result.failure(...
                    grey.error.GreyError.validation('User ID is required'));
                return;
            end
            
            path = sprintf('/users/%s', userId);
            result = obj.httpClient.get(path);
        end
        
        function result = getCurrent(obj)
            %GETCURRENT Get the current authenticated user
            %   result = client.getCurrent()
            
            result = obj.httpClient.get('/users/me');
        end
    end
end
